### DISCUSSION TITLE
Multi search

### DISCUSSION THREAD

**Discussion by vikiival** [kodadot/nft-gallery#4457] [General]
Hey,
in upcoming days I plan to build first version of #4257.

Here is my question: *What kind of data would you like to see/ have in the search for easy and quick development?*

cc:
@kodadot/code-review-guild 
@exezbcz

**Comment by spacesmutje** [comment]
Chain/Recently listed/Most interactions/Price in fiat/respective asset

Or did i get the question completely wrong 😂

**Comment by roiLeo** [comment]
> Hey, in upcoming days I plan to build first version of #4257.

Can we have an idea of the solution chosen? What's the stack?
Is it going to wrap all gql endpoint? 
Are we going to use `uniquery`?
How you'll handle different chain spec? like filter Offers on BSX or sorting by emotes on RMRK

> _What kind of data would you like to see/ have in the search for easy and quick development?_

🤔 json data?

![am-iright-anchorman](https://user-images.githubusercontent.com/9987732/206706699-7ca87f26-0f99-47b2-8be1-2a0818a32f08.gif)

**Reply by vikiival** [reply]
> stack

We decided to go with supabase pro 
With graphql enabled. 
This way we would be quite flexible on the 

> how it would work 

The critical/static data - name, various ids, flatten metadata and chain prefix would be inside the supabase db

via graphql we can fetch the data

> uniquery

advantage od subapase are edge functions therefore we can build our own uniquery like stuff

**Comment by exezbcz** [comment]
yup, so for the multichain search, originally there was this draft of search results/trending collection. These are the data that should be shown
It consists of: units, owner, floor etc. 
- this is solely search
screens:
![image](https://user-images.githubusercontent.com/90852205/206764512-f585a98a-8d43-455e-b62a-bd83fd4605aa.png)
![image](https://user-images.githubusercontent.com/90852205/206764542-6a477f46-2ae8-4b5c-9e0a-663722952e3f.png)

**Comment by vikiival** [comment]
So I managed to pushed the data into `Supabase`
If anyone wants to play with that here is the URL

```
https://xtnvtbdyvpbbnonbbcif.supabase.co/graphql/v1
```

You need to add `apiKey` header into your request. You can find the key in the discord group.


Sample Query
```graphql
query findCollections {
	collectionsCollection(first: 10) {
		edges {
			node {
				name
			}
		}
	}
}
```

**Comment by vikiival** [comment]
So I finally iterated to the point where search can be usable

to use a search you need to make request to `https://xtnvtbdyvpbbnonbbcif.functions.supabase.co/search`

values that you always need to provide are `table` and `search`

You have two alternatives:

### 1. use GET

```
https://xtnvtbdyvpbbnonbbcif.functions.supabase.co/search?table=collections&search=ksm
```


### 2. use POST

```
https://xtnvtbdyvpbbnonbbcif.functions.supabase.co/search

BODY:
{
  "table": "collections",
  "search": "ksm"
}
```

additional fields that you can use for search:

```typescript
type SearchQuery = {
  table: 'collections' | 'nfts',
  select: string,
  search: string,
  issuer?: string,
  owner?: string,
  chain?: string,
  limit: number,
  offset: number,
}
```

Note there is a current limit hardcoded limit 25 items

I also [shared the anon key for you in our discord](https://discord.com/channels/840514076538830888/842717226758701056/1062726371907993723)


sample request

```
curl -L -X POST 'https://xtnvtbdyvpbbnonbbcif.functions.supabase.co/search' -H 'Authorization: Bearer REPLACE_WITH_TOKEN_FROM_DC' --data '{"table":"collections", "search": "viki"}'
```

**Reply by vikiival** [reply]
Just a note that database does not contain all data yet
I am working to bring it to the realtime

**Reply by vikiival** [reply]
Also please let me know what features do you want to see in the search.

[I will open-source the repo soon](https://github.com/kodadot/polysearch)

**Reply by preschian** [reply]
anyway, how to get access to that discord?

**Reply by vikiival** [reply]
It is kodadot discord channel 
Please ping me a message on dc

**Comment by exezbcz** [comment]
👀
